DIFixture/DependencyInjectionFixture.cs

```diff
@@ -28,6 +28,18 @@ public void RegisterTypeOnlyByInterface_ThrowsArgumentException()
             Assert.Throws<ArgumentException>(() => container.Register<IMessagePrinter>(ServiceLifetime.Singleton));
         }
 
+        [Test]
+        public void RegisterAndResolveAfterResolving_ShouldResolveTwoServices()
+        {
+            container.Register<IMessagePrinter, FileMessageWriter>(ServiceLifetime.Singleton);
+            container.Register(new string("message"), ServiceLifetime.Singleton);
+            var fileMesWriterResolveResult = container.Resolve<FileMessageWriter>();
+            container.Register<IMessagePrinter, ConsoleMessageWriter>(ServiceLifetime.Singleton);
+            var consoleMesWriterResolveResult = container.Resolve<ConsoleMessageWriter>();
+            Assert.That(fileMesWriterResolveResult, Is.Not.EqualTo(null));
+            Assert.That(consoleMesWriterResolveResult, Is.Not.EqualTo(null));
```

Comment:
Verification looks weird, I don't think that if you put candy in a basket you'll expect to get NOT a dog from there